### ISSUE TITLE
MediaField: BagPart edit, Content type resource loading system create an unnecessary `script` tag

### ISSUE THREAD

**Issue by PiemP** [OrchardCMS/OrchardCore#12868]
### Describe the bug

This is not something serious but could cause some problems in future.

In the backend interfaces, if you create a content type with a `BagPart` that could contains items with a `MediaField`, the system that loads all the resources related to the content type contained in the `BagPart` cause the generation of a `script` tag that is unnecessary because is not related tro any DOM elements and the necessary `script` tag are loaded and correclty settedup when you add in item to the `BagPart`. 

### To Reproduce
1.  Create a ContentType with a MediaField
2.  Create a ContentType with a BagPart that could contains elements of the ContentType created before
3.  Go to insert a new ContentItem of the ContentType created in the previous poin
4.  Open the browser developer tools and search something similar to the content of the screenshot (see below, where `Image` is the name of the ContentType and Media is the technical name of the `MediaField`)

### Expected behavior
The BagPart's resource loading system have to load only specific resources or avoid to load resources not loaded from a specific file.

Thank you.

### Screenshots
![Screenshot 2022-11-21 154134](https://user-images.githubusercontent.com/2736009/203304668-d8caf9be-e6b5-41f9-b799-c5ce17348aec.png)

**Comment by sebastienros** [comment]
Do you mean that it's expected to have all the scripts for all the parts executed but in this case this script is useless because it's for a non-exissting dom element? (the id varies for each item).

**Comment by PiemP** [comment]
> Do you mean that it's expected to have all the scripts for all the parts executed but in this case this script is useless because it's for a non-exissting dom element? (the id varies for each item).

Yes, I know it`s a little thing. The DOM element connected to the script tag created on the page load doesn't exist. This make it useless. When you add a Content Item to the bagpart, during an edit operation, the "real" script related to the item initialization is in the template loaded by the bagpart's code and correctly executed.

I have tried to understand how the system works in this case: for now, I don't understand how to solve it. 

Probably the easiest way to solve this little thing is to load only the resources that coming from a file because generally, this kind of code, is not subject of a dynamic behavior. 

The script tag loading could be leaved to the code that loads the item's template when you add a Contente Item to the bagpart. The last thing is already done in some way. Remains to implement the first part of the request.

**Comment by Skrypt** [comment]
I started refactoring the media gallery with Vue 3. We need to have something done better for these Vue.js apps/components to load them dynamically. @jptissot

**Comment by gvkries** [comment]
This behavior is intentional and not limited to the media field. Any script associated with a content type that can be added to the bag part is also included with an arbitrary HTML ID before a contained content item is created.

However, since HTML ID prefixes are generated from a Guid, the first character is replaced with a z if it's a digit. As a result, the original ID cannot be matched in the DOM.
